repair/repair.cc

```diff
@@ -1218,13 +1217,12 @@ struct repair_options {
             options.erase(it);
         }
     }
-
+public:
     // A range is expressed as start_token:end token and multiple ranges can
     // be given as comma separated ranges(e.g. aaa:bbb,ccc:ddd).
-    static void ranges_opt(dht::token_range_vector& var,
-            std::unordered_map<sstring, sstring>& options,
-                        const sstring& key) {
-        auto it = options.find(key);
+    void set_ranges(std::unordered_map<sstring, sstring>& options,
+            dht::i_partitioner& partitioner) {
+        auto it = options.find(RANGES_KEY);
         if (it == options.end()) {
```

Comment:
The changelog says "repair: stop calling global_partitioner()" but that's not what I see here.

Reply:
Right. I don't understand this patch at all. Also, @asias please review this patch that touches repair in non-obvious (to me) ways.

Reply:
I added some comments.

Reply:
This change is gone once I moved the check to repair_info constructor.